pandas/tseries/base.py

```diff
@@ -67,7 +67,7 @@ def __contains__(self, key):
         try:
             res = self.get_loc(key)
             return np.isscalar(res) or type(res) == slice or np.any(res)
-        except (KeyError, TypeError):
+        except (KeyError, TypeError, ValueError):
```

Comment:
I think this is a reasonable patch, but ultimately the cleaner patch would be to fix `_is_convertible_to_td` to not fail on strings like `'foo'` (or better yet, use exception handling instead of separately asking if the operation is valid).